src/qibolab/platforms/platform.py

```diff
@@ -1,6 +1,94 @@
+from dataclasses import asdict, dataclass, replace
+from enum import Enum, auto
+from typing import Optional
+
 from qibo.config import raise_error
 
 from qibolab.platforms.abstract import AbstractPlatform
+from qibolab.result import (
+    AveragedIntegratedResults,
+    AveragedRawWaveformResults,
+    AveragedStateResults,
+    IntegratedResults,
+    RawWaveformResults,
+    StateResults,
+)
+
+
+class AcquisitionType(Enum):
+    """
+    Types of data acquisition from hardware.
+
+    SPECTROSCOPY: Zurich Integration mode for RO frequency sweeps,
+    INTEGRATION: Demodulate and integrate the waveform,
+    RAW: Acquire the waveform as it is,
+    DISCRIMINATION: Demodulate, integrate the waveform and discriminate among states based on the voltages
+
+    """
+
+    RAW = auto()
+    INTEGRATION = auto()
+    DISCRIMINATION = auto()
+    SPECTROSCOPY = auto()
+
+
+class AveragingMode(Enum):
+    """
+    Types of data averaging from hardware.
+
+    CYLIC: Better averaging for noise,
+    SINGLESHOT: False averaging,
+    [SEQUENTIAL: Worse averaging for noise]
+
+    """
+
+    CYCLIC = auto()
+    SINGLESHOT = auto()
+    SEQUENTIAL = auto()
+
+
+RESULTS_TYPE = {
+    AveragingMode.CYCLIC: {
+        AcquisitionType.INTEGRATION: AveragedIntegratedResults,
+        AcquisitionType.RAW: AveragedRawWaveformResults,
+        AcquisitionType.DISCRIMINATION: AveragedStateResults,
+    },
+    AveragingMode.SINGLESHOT: {
+        AcquisitionType.INTEGRATION: IntegratedResults,
+        AcquisitionType.RAW: RawWaveformResults,
+        AcquisitionType.DISCRIMINATION: StateResults,
+    },
+}
+
+
+@dataclass(frozen=True)
+class ExecutionParameters:
+    """Data structure to deal with execution parameters
+
+    :nshots: nshots (int): Number of shots to sample from the experiment. Default is 1024.
+    relaxation_time (int): Time to wait for the qubit to relax to its ground state between shots in s.
+                If ``None`` the default value provided as ``relaxation_time`` in the runcard will be used.
+    :fast_reset (bool): Enable or disable fast reset
+    :acquisition_type (AcquisitionType): Data acquisition mode
+    :averaging_mode (AveragingMode): Data averaging mode
+    """
+
+    nshots: Optional[int] = None
+    relaxation_time: Optional[int] = None
+    fast_reset: bool = False
+    acquisition_type: AcquisitionType = AcquisitionType.DISCRIMINATION
+    averaging_mode: AveragingMode = AveragingMode.SINGLESHOT
+
+    def __post_init__(self):
+        if not isinstance(self.acquisition_type, AcquisitionType):
+            raise TypeError("acquisition_type is not valid")
```

Comment:
Here I would elaborate on the error by showing which wrong type the user was using.